requests/models.py

```diff
@@ -400,6 +400,11 @@ def prepare_headers(self, headers):
 
         if headers:
             self.headers = CaseInsensitiveDict((to_native_string(name), value) for name, value in headers.items())
+            # Explicitly convert any non-string header value to its string
+            # representation.
+            for key, val in self.headers.items():
+                if not isinstance(val, basestring):
+                    self.headers[key] = str(val)
```

Comment:
This is the pretty close to the right place, but not really right. It'd be better if we could do it in the generator comprehension above.

Basically, wrap the `isinstance` block up in a function and put it in `requests.utils`. Probably call it something like `convert_to_string`. Then just call it on each value in the generator comprehension on L402.